src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts

```
@@ -0,0 +1,351 @@
+import '@testing-library/jest-dom'
+
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { act } from 'react-dom/test-utils'
+import { h } from 'react-hyperscript-helpers'
+import { AzureStorage, AzureStorageContract } from 'src/libs/ajax/AzureStorage'
+import { GoogleStorage, GoogleStorageContract } from 'src/libs/ajax/GoogleStorage'
+import { defaultAzureWorkspace, defaultGoogleWorkspace } from 'src/pages/workspaces/workspace/analysis/_testData/testData'
+import {
+  AbsolutePath, AnalysisFile, getDisplayName, getExtension, getFileName
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import {
+  AnalysisDuplicator,
+  AnalysisDuplicatorProps
+} from 'src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator'
+import { cloudProviders } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+import {
+  getToolFromFileExtension,
+  ToolLabel,
+  toolLabelTypes,
+  tools
+} from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { cloudProviderTypes } from 'src/pages/workspaces/workspace/workspace-utils'
+
+
+jest.mock('src/components/Modal', () => {
+  const { div, h } = jest.requireActual('react-hyperscript-helpers')
+  const originalModule = jest.requireActual('src/components/Modal')
+  return {
+    ...originalModule,
+    __esModule: true,
+    default: props => div({ id: 'modal-root' }, [
+      h(originalModule.default, { onAfterOpen: jest.fn(), ...props })
+    ])
+  }
+})
+
+jest.mock('src/libs/notifications', () => ({
+  notify: (...args) => {
+    console.debug('######################### notify')/* eslint-disable-line */
+    console.debug({ method: 'notify', args: [...args] })/* eslint-disable-line */
```

Is this debug code needed anymore?

---

its helpful when tests fail, as it allows the error reporting to print it. We need to mock this function anyways, so I figure its helpful to mock it as something that doesn't eat any error messaging.